src/common/snippets/tests/include/lir_comparator.hpp

```diff
@@ -75,6 +75,9 @@ class LIRComparator {
     static Result compare_loop_managers(const ov::snippets::lowered::LoopManagerPtr& loop_manager,
                                         const ov::snippets::lowered::LoopManagerPtr& loop_manager_ref);
 
+    static Result compare_loop_managers_without_index(const ov::snippets::lowered::LoopManagerPtr& loop_manager,
```

Comment:
We have a special pass that normalizes loop ids [here](https://github.com/openvinotoolkit/openvino/blob/013ed70e1a53e447a694f31ce519cd9f89d2ea0b/src/common/snippets/include/snippets/lowered/pass/normalize_loop_ids.hpp#L14-L30).

Can we just add it to the test pipeline instead of introducing a very peculiar comparator that's apparently needed only for one pass?

Reply:
Regarding Loop expressions creation: `linear_ir->push_node` indeed can't be used in this case, since we need to provide some LIR specific information (e.g. port connectors) for loops creation. However, it looks like we can use `insert_node` API for loop exprs creation. For instance, `LoopBegin` creation:
```
        const auto loop_begin = std::make_shared<ov::snippets::op::LoopBegin>();
        const auto loop_begin_expr = linear_ir->insert_node(loop_begin, std::vector<PortConnectorPtr>{}, {}, false, linear_ir->end());
```

Reply:
@IvanNovoselov @v-Golubev , This is done in [96b25cc](https://github.com/openvinotoolkit/openvino/pull/24044/commits/96b25cc7e4c65347dedf509de7f1302bf56da4ac). To not get LIR complex, I create a separate test class `ExtractLoopInvariantsRemoveLoopsTest`, where `NormalizeLoopIDs` pass is applied and put test case removing loops under it(which have explicit loopBegin and loopEnd in LIR). Thank you very much for the suggestions :-)

Reply:
@chenhu-wang BTW feel free to create a ticket if you think that the functionality of the `NormalizeLoopIDs` pass is insufficient or the pass is inconvenient to use. Just let us know about your experience in the teams chat: "Guys I think that this pass doesn't really work good because of this and that. Here is the ticket with the problem description."
And we will try improve or generalize it.